app/src/main/python/gogdl/dl/dl_utils.py

```
@@ -0,0 +1,184 @@
+def get_zlib_encoded(api_handler, url):
+    retries = 5
+    while retries > 0:
+        try:
+            x = api_handler.session.get(url, timeout=TIMEOUT)
+            if not x.ok:
+                return None, None
+            try:
+                decompressed = json.loads(zlib.decompress(x.content, 15))
+            except zlib.error:
+                return x.json(), x.headers
+            return decompressed, x.headers
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)